torchvision/prototype/datasets/_builtin/pcam.py

```diff
@@ -0,0 +1,122 @@
+import io
+from torchvision.prototype import features
+from typing import Any, Callable, Dict, List, Optional, Tuple, Iterator
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe, Mapper, Zipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    OnlineResource,
+    DatasetType,
+    GDriveResource,
+)
+from torchvision.prototype.datasets.utils._internal import (
+    hint_sharding,
+    hint_shuffling,
+)
+from torchvision.prototype.features import Label
+
+
+class H5Reader(IterDataPipe[Tuple[str, io.IOBase]]):
+    def __init__(
+        self,
+        datapipe: IterDataPipe[Tuple[str, io.IOBase]],
+        key: Optional[str] = None,  # Note: this key thing might be very specific to the PCAM dataset
```

Comment:
Nit: I would reserve line comments for linter directives.

```suggestion
        # Note: this key thing might be very specific to the PCAM dataset
        key: Optional[str] = None,
```

This applies to all occurrences of this.

Reply:
Do you feel strongly about this? I like line comments because it's very clear what they apply to.
In some occasions, I have a hard time figuring out whether a comment applies to a single line below, or to a bigger block

Reply:
No strong opinion.